sympy/matrices/immutable.py

```diff
@@ -199,5 +185,27 @@ def as_immutable(self):
     def is_diagonalizable(self, reals_only=False, **kwargs):
         return super().is_diagonalizable(
             reals_only=reals_only, **kwargs)
+
     is_diagonalizable.__doc__ = SparseMatrix.is_diagonalizable.__doc__
     is_diagonalizable = cacheit(is_diagonalizable)
+
+
+@dispatch(ImmutableDenseMatrix, MatrixExpr)
+def _eval_Eq(lhs, rhs): # noqa:F811
+    return None
```

Comment:
Is this handler needed? Can it not fall back to a general `Basic, Basic` handler that returns None?